### ISSUE TITLE
LabeledPrice array must be json-encoded in SendInvoice method

### ISSUE THREAD

**Issue by ircop** [unreal4u/telegram-api#32]
Subj.
It's not obvious that array of LabeledPrice objects must be json_encode`d when passing to SendInvoice.

```
$invoice = new SendInvoice();
$price = new LabeledPrice( ['amount' => 100, 'label'=>'usd'] );
$invoice->prices = [ $price ];
...
$tg->performApiRequest( $invoice );
```

This code will result an error from Telegram API:
{"ok":false,"error_code":400,"description":"Bad Request: prices are not specified"}

To call this method successfully, we must json_encode "prices" parameter:
```
$invoice->prices = \GuzzleHttp\json_encode([ $price ]);
```
This way all working good.

I think the best solution would be some detection in SendInvoice: if data passed as array, encode data to json, else do nothing

**Comment by unreal4u** [comment]
Oh nice catch! I haven't been able to test this personally, so the whole payment section may contain some errors. 

Will fix this tonight :) Thanks for reporting!

**Comment by unreal4u** [comment]
Actually the fix is pretty simple: make a `performSpecialConditions()` function which can perform this operation. This is defined by the abstract: 
https://github.com/unreal4u/telegram-api/blob/fe93f5c63fe27ee0959cb9d25ac49e0ed27d430e/src/Abstracts/TelegramMethods.php

And can be extended to cover use-cases such as this one.

Will implement it tonight ;)

**Comment by ircop** [comment]
Thanks for the quick response! :)

**Comment by unreal4u** [comment]
This is fixed in version 2.4.2: https://github.com/unreal4u/telegram-api/releases/tag/v2.4.2

Thanks for reporting this one! Success with your bot :)
